File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        ans = []
        map = {}


        for i in range(0,len(nums)):
            diff = target - nums[i]

            if diff in map:
                ans.append(i)
                ans.append(map[diff])
                break
            
            map[nums[i]] = i

        return ans    

File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer,Integer> mp1 = new HashMap<>();
        int[] ans = new int[2];

        for(int i =0;i<nums.length;i++)
        {
            int diff = target - nums[i];

            if(mp1.containsKey(diff))
            {
                ans[0] = i;
                ans[1] = mp1.get(diff);
            }

            mp1.put(nums[i], i);
        }

        return ans;
    }
}